### ISSUE TITLE
no-unused-variable without use of `this`

### ISSUE THREAD

**Issue by leocaseiro** [palantir/tslint#2790]
### Bug Report

Hi, I believe there's an error with the `no-unused-variable` when not using with `this`.

You can see my example in Angular with constructor when I'm not using `this`;

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.3.2
- __Running TSLint via__: Angular CLI (ng lint)

#### TypeScript code being linted

```ts
@Component({
})
class MyComponent {
    constructor(private myService: MyService) {
        myService.method(); // works on TS but fail on tslint
        // this.myService.method(); // tslint pass
    }
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-unused-variable": true
    }
}
```

#### Actual behavior
`Property 'myService' is declared but never used.`

#### Expected behavior
`Should pass`

**Comment by ajafff** [comment]
The behavior is correct. The `private` modifier creates a property with the same name as the parameter. 
You're only using the parameter and not the property. Hence the error message `Property myService is declared but never used.`
